src/mon/Paxos.cc

```diff
@@ -1393,6 +1393,11 @@ void Paxos::peon_init()
 
 void Paxos::restart()
 {
+  if (is_shutdown()) {
```

Comment:
this alleviates the symptom of the racing.  but the racing is not fixed. i think one way to fix it is to delete the loggers in destructor.

Reply:
Could a similar race happen on Monitor::logger? It's very similar in its usage.

Reply:
yeah, very likely! at least i cannot find any lock to protect Monitor from this kind of racing after glimpsing through `Monitor.cc`.

Reply:
I'll organise a similar PR for the Monitor code, thanks!